parachain/pallets/omni-account/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub(crate) mod identity_context;
+
+pub use core_primitives::{assertion::network::Web3Network, Identity, OmniAccountIdentity};
+pub use frame_system::pallet_prelude::BlockNumberFor;
+pub use pallet::*;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use identity_context::*;
+use sp_core::H256;
+use sp_core_hashing::blake2_256;
+use sp_std::vec::Vec;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The event type of this pallet.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The origin which can manage the pallet.
+		type TEECallOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// The maximum number of identities an id graph can have.
+		#[pallet::constant]
+		type MaxIDGraphLength: Get<u32>;
+	}
+	pub type IDGraphLinks<T> =
+		BoundedVec<(OmniAccountIdentity, IdentityContext<T>), <T as Config>::MaxIDGraphLength>;
+
+	#[pallet::storage]
+	pub type LinkedIdentities<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = H256, Value = ()>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn id_graphs)]
+	pub type IDGraphs<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = Identity, Value = IDGraphLinks<T>>;
```

Comment:
I favor `T::AccountId` over `Identity` in this case - it shows explictly that the OmniAccount is a native substrate account under the hood.

So whichever Identity you use as a prime key to log onto our service, we'll derive a substrate account for you (and show it in the UI), in case users want to check balance/lookup tx their should use this account, our tooling UI can help with that.

One more thing - users can even unlink their prime identity after the OmniAccount (if we support the remove_identity UserOp) is set and only use MemberIdentities to control the OmniAccount

Reply:
We'll need another storage, either in `IDGraphs` or as a separate storage to record:
- in which block number the OmniAccount is created
- the whole IDGraph fingerprint (hash)

This can be done in another PR though